source/npm/util.js

```diff
@@ -159,7 +159,10 @@ async function getFilesIgnoredByDotnpmignore(pkg, fileList) {
 		path: pkgDir.sync(),
 		ignoreFiles: ['.npmignore']
 	});
-	return fileList.filter(minimatch.filter(getIgnoredFilesGlob(allowList, pkg.directories), {matchBase: true, dot: true}));
+	// We need to filter the allowlist because it includes too many files (including .git and node_modules)
+	// Ideally, we would filter some directories directly with `ignoreWalker`.
+	const filteredAllowList = allowList.filter(minimatch.filter(`!{${filesIgnoredByDefault.join(',')}}`, {matchBase: true, dot: true}));
```

Comment:
Dynamically creating a glob pattern like this feels fragile. I think you at least need some kind of escaping.